resources/splunk_install.rb

```diff
@@ -0,0 +1,219 @@
+    end
+  end
+
+  def install_dir
+    @install_dir ||= default_install_dirs[package][node['os'].to_sym]
+    raise "Unsupported Combination: #{package} + #{node['os']}" unless @install_dir
+    @install_dir
+  end
+
+  def package_url
+    @package_url ||=
+      case package
+      when :splunk then "#{source_url}/splunk/releases/#{version}/#{node['os']}/splunk"
+      when :universal_forwarder then "#{source_url}/universalforwarder/releases/#{version}/#{node['os']}/splunkforwarder"
+      end + "-#{version}-#{build}-#{kernel_string}"
+  end
+
+  def package_path
+    @package_path ||= Pathname.new(Chef::Config['file_cache_path']) + filename_from_url(package_url)
+  end
+
+  ### Inherited Actions
+
+  load_current_value do |desired|
+    package desired.package
+    version_file = version_pathname
+    current_value_does_not_exist! unless version_file.exist?
+
+    version_data = version_file.read
+    installed_version, installed_build = version_data.match(/VERSION=(\d(?:\.\d)+).+BUILD=([\w]+)/m).captures
+    version installed_version
+    build installed_build
+  end
+
+  action :install do
+    state = node.run_state['splunk_ingredient'] ||= {}
+    state['packages'] ||= {}
+
+    raise "Installation resource with name #{name} already exists!" if state['packages'][name]
+    state['current_package'] = state['packages'][name] = {
+      package: package,
+      version: version,
+      build: build,
+      x64: x64_support
+    }
+
+    converge_if_changed :version, :build do
+      remote_file package_path.to_s do
+        source package_url
+        show_progress true if defined? show_progress # Chef 12.9 feature
+        notifies :delete, "remote_file[#{package_path}]", :delayed
+      end
+    end
+
+    user service_user do
+      password service_password if service_password
+      system true
+      action :create
+    end
+  end
+
+  action :uninstall do
+    directory install_dir do
+      recursive true
+      action :delete
+    end
+  end
+end
+
+###################################
+### Platform Specific Providers ###
+###################################
+
+class LinuxInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, os: 'linux'
+
+  def kernel_string
+    x64_support ? 'Linux-x86_64.tgz' : 'Linux-i686.tgz'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      tar_extract package_path do
+        action :extract_local
+        target_dir install_dir
+      end
+    end
+  end
+end
+
+class RedhatInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, platform_family: 'rhel'
+
+  def kernel_string
+    x64_support ? 'linux-2.6-x86_64.rpm' : 'i386.rpm'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      rpm_package package_name do
+        source package_path.to_s
+        action :install
+      end
+    end
+  end
+
+  action :uninstall do
```

Comment:
I ended up super()ing anyways to share run_state changes
c2e8a3bdf6cd33e9779c4cc76d0802e600738ca9